tests/def_list.md

```diff
@@ -10,13 +10,122 @@ Second Term
 which is multiline
 
 : This is not a correct definition list
+
+# With multiple paragraphs
+
+First Term
+: This is the definition of the first term.
+
+Second Term
+: This is one paragraph of the second term.
+
+  This is another paragraph of the second term.
+
+Third term
+: This is one paragraph using lazy wrapping,
+so this should be part of the first paragraph
+of the third term.
+
+  But this should be the second paragraph of
+  the second paragraph
+
+  Note that, currently, subsequent paragraphs cannot lazy wrap.
+For example, this line will be part of a new paragraph outside
+of the definition list entirely.
+
+Fourth term
+: > There is a block quote in the 1st definition of the fourth term.
+: But not in the second.
+
+# Nesting
+
+Root
+: Level one
+
+  Nested
+  : Level two
+    ```
+    code
+    ```
+
+    More
+    : Level three
+
+Root again
+: Level one again
+
+# A heading
+: Followed by a line beginning with a `:` is not a definition.
+
 .
 <dl><dt>First Term</dt>
-<dd>This is the definition of the first term.</dd>
+<dd>
+This is the definition of the first term.
+</dd>
 <dt>Second Term</dt>
-<dd>This is one definition of the second term.</dd>
-<dd>This is another definition of the second term.
-which is multiline</dd>
+<dd>
+This is one definition of the second term.
+</dd>
+<dd>
+This is another definition of the second term.
+which is multiline
+</dd>
 </dl>
 <p>: This is not a correct definition list</p>
+<h1>With multiple paragraphs</h1>
+<dl><dt>First Term</dt>
+<dd>
+<p>This is the definition of the first term.</p>
+</dd>
+<dt>Second Term</dt>
+<dd>
+<p>This is one paragraph of the second term.</p>
+<p>This is another paragraph of the second term.</p>
+</dd>
+<dt>Third term</dt>
+<dd>
+<p>This is one paragraph using lazy wrapping,
+so this should be part of the first paragraph
+of the third term.</p>
+<p>But this should be the second paragraph of
+the second paragraph</p>
+<p>Note that, currently, subsequent paragraphs cannot lazy wrap.</p>
+</dd>
+</dl>
+<p>For example, this line will be part of a new paragraph outside
+of the definition list entirely.</p>
```

Comment:
@talex5 does this seem like the right behavior to you? I'm not exactly sure what ought to happen in this case. I can report that in pandoc, this markdown

```
Term 1
: A first paragraph that is
lazily wrapped

    A second paragrap that 
is lazily wrapped 
```

is rendered as 

```html
<dl>
<dt>Term 1</dt>
<dd><p>A first paragraph that is lazily wrapped</p>
<p>A second paragrap that is lazily wrapped</p>
</dd>
</dl>
```

But they use 4 spaces to indicate the start of the second paragraph.